src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,299 @@
+ * value. Collected edges are within some linear search area (configurable).
+ *
+ * @author seancoulter
+ */
+class RoadNameSpellingConsistencyCheckWalker extends EdgeWalker
+{
+
+    /**
+     * Directional character, sometimes appended to start/end of road name (e.g. Banana Ave W)
+     */
+    enum Direction
+    {
+        N,
+        S,
+        E,
+        W
+    }
+
+    private static final EnumSet<Direction> DIRECTIONS = EnumSet.of(Direction.N, Direction.S,
+            Direction.E, Direction.W);
+
+    private static final int NINE = 9;
+
+    private static final int ZERO = 0;
+
+    // ASCII character to digit offset
+    private static final int ASCII_OFFSET = 48;
+
+    /**
+     * Evaluate the {@link org.openstreetmap.atlas.tags.names.NameTag}s of the startingEdge and an
+     * incomingEdge to see if their spellings are inconsistent with one another.
+     *
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumAllowedDifferences
+     *            the number of Levenshtein edits allowed
+     * @return true if incomingEdge's name is at most the desired Levenshtein distance from the
+     *         start edge's name
+     */
+    static Predicate<Edge> isEdgeWithInconsistentSpelling(final Edge startEdge,
+            final int maximumAllowedDifferences)
+    {
+        return incomingEdge ->
+        {
+            final int similarityIndex = similarityIndex(incomingEdge, startEdge);
+            return similarityIndex <= maximumAllowedDifferences && similarityIndex > 0;
+        };
+    }
+
+    /**
+     * Determine the cost of substituting two characters in the Levenshtein distance calculation.
+     *
+     * @param first
+     *            a character to examine for equality with second
+     * @param second
+     *            a character to examine for equality with first
+     * @return 0 if first & second are equal, 1 otherwise
+     */
+    private static int costOfSubstitution(final char first, final char second)
+    {
+        return first == second ? 0 : 1;
+    }
+
+    /**
+     * The function used to collect {@link Edge}s that fall within the search area.
+     *
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumSearchDistance
+     *            the maximum distance from the end of the incoming edge to the start of the
+     *            starting edge
+     * @return A stream of edges that fall in the search area
+     */
+    private static Function<Edge, Stream<Edge>> edgesWithinMaximumSearchDistance(
+            final Edge startEdge, final Distance maximumSearchDistance)
+    {
+        return incomingEdge -> incomingEdge.end().getLocation()
+                .distanceTo(startEdge.start().getLocation())
+                .isLessThanOrEqualTo(maximumSearchDistance)
+                        ? incomingEdge.connectedEdges().stream().filter(Edge::isMasterEdge)
```

Comment:
Do we need connected edges with no name tags? If not, we can filter them here.

Reply:
Made that change, but reverted it because it was dropping some TPs. I'm a bit surprised by that since it looks like it's only a difference in _where_ EdgeWalker filters these Edges without names.